### DISCUSSION TITLE
Retrieve all pages and filter them by category

### DISCUSSION THREAD

**Discussion by just-alexis** [sulu/sulu#7574] [Q&A] [answered]
Hello,
I want to make a blog homepage, For this i need to retrieve all pages with template "articles" and add a pagination.
I did that easily with a smart content.

But i want to add 3 "filter buttons" like "All category", "Category 1" and "Category 2".
When I click on a button I want it to filter the associated pages and keep the pagination up. How can i do this ? 
I have to add custom javascript to filter the smart content result ? But i dont know if it will break the pagination ?

Or i have to add multiple smart content that i display or hide when I click on a button ?

Also I don't use the blog bundle because I can't install elastic search.

Thx for your help

**Comment by alexander-schranz** [comment, marked as answer]
> But i want to add 3 "filter buttons" like "All category", "Category 1" and "Category 2".

Smart content support out of the box filtering by categories and tags via query parameterson the website. You even can control the query parameter via categories_parameter as documented on https://docs.sulu.io/en/2.6/reference/content-types/smart_content.html

To create filter links have a look at the category twig extensions: https://docs.sulu.io/en/2.6/reference/twig-extensions/functions/sulu_categories.html / https://docs.sulu.io/en/2.6/reference/twig-extensions/index.html#categorybundle

PS: We currently working on SuluArticleBundle 3.0 which does not longer require elasticsearch for articles. The current state can be found in the `^3.0@dev` version.

**Comment by just-alexis** [comment]
Thank you, that's exactly what I was looking for. I have an additional question: is it possible to also search for articles with this title or description using a text field?

**Reply by alexander-schranz** [reply]
articles have an own index which can also be searched for. for complexer queries you can even use directly the elasticsearch client which we do for projects like our case study here: https://sulu.io/case-studies/kuechengoetter
